cumulus/parachains/runtimes/collectives/collectives-westend/src/ambassador/mod.rs

```diff
@@ -0,0 +1,256 @@
+	pub AmbassadorPalletAccount: AccountId = account::AMBASSADOR_REFERENDA_PALLET_ID.into_account_truncating();
+}
+
+pub type AmbassadorReferendaInstance = pallet_referenda::Instance2;
+
+impl pallet_referenda::Config<AmbassadorReferendaInstance> for Runtime {
+	type WeightInfo = weights::pallet_referenda_ambassador_referenda::WeightInfo<Runtime>;
+	type RuntimeCall = RuntimeCall;
+	type RuntimeEvent = RuntimeEvent;
+	type Scheduler = Scheduler;
+	type Currency = Balances;
+	// A proposal can be submitted by a member of the Ambassador Program of
+	// [ranks::SENIOR_AMBASSADOR_TIER_3] rank or higher.
+	type SubmitOrigin = pallet_ranked_collective::EnsureMember<
+		Runtime,
+		AmbassadorCollectiveInstance,
+		{ ranks::SENIOR_AMBASSADOR_TIER_3 },
+	>;
+	type CancelOrigin = EitherOf<EnsureRoot<AccountId>, EnsureHeadAmbassadorsVoice>;
+	type KillOrigin = EitherOf<EnsureRoot<AccountId>, EnsureHeadAmbassadorsVoice>;
+	type Slash = ToParentTreasury<WestendTreasuryAccount, AmbassadorPalletAccount, Runtime>;
+	type Votes = pallet_ranked_collective::Votes;
+	type Tally = pallet_ranked_collective::TallyOf<Runtime, AmbassadorCollectiveInstance>;
+	type SubmissionDeposit = SubmissionDeposit;
+	type MaxQueued = ConstU32<20>;
+	type UndecidingTimeout = UndecidingTimeout;
+	type AlarmInterval = AlarmInterval;
+	type Tracks = tracks::TracksInfo;
+	type Preimages = Preimage;
+}
+
+parameter_types! {
+	pub const AnnouncementLifetime: BlockNumber = 180 * DAYS;
+	pub const MaxAnnouncements: u32 = 50;
+}
+
+pub type AmbassadorContentInstance = pallet_collective_content::Instance1;
+
+impl pallet_collective_content::Config<AmbassadorContentInstance> for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type CharterOrigin = EitherOf<EnsureRoot<AccountId>, EnsureHeadAmbassadorsVoice>;
+	type AnnouncementLifetime = AnnouncementLifetime;
+	// An announcement can be submitted by a Senior Ambassador member or an ambassador plurality
+	// voice taken via referendum.
+	type AnnouncementOrigin = EitherOfDiverse<
+		pallet_ranked_collective::EnsureMember<
+			Runtime,
+			AmbassadorCollectiveInstance,
+			{ ranks::SENIOR_AMBASSADOR_TIER_3 },
+		>,
+		EnsureAmbassadorsVoice,
+	>;
+	type MaxAnnouncements = MaxAnnouncements;
+	type WeightInfo = weights::pallet_collective_content::WeightInfo<Runtime>;
+}
+
+pub type AmbassadorCoreInstance = pallet_core_fellowship::Instance2;
+
+impl pallet_core_fellowship::Config<AmbassadorCoreInstance> for Runtime {
+	type WeightInfo = weights::pallet_core_fellowship_ambassador_core::WeightInfo<Runtime>;
+	type RuntimeEvent = RuntimeEvent;
+	type Members = pallet_ranked_collective::Pallet<Runtime, AmbassadorCollectiveInstance>;
+	type Balance = Balance;
+	// Parameters are set by any of:
+	// - Root;
+	// - the FellowshipAdmin origin (i.e. token holder referendum);
+	// - a vote among all Head Ambassadors.
+	type ParamsOrigin = EitherOfDiverse<
+		EnsureXcm<IsVoiceOfBody<GovernanceLocation, FellowshipAdminBodyId>>,
```

Comment:
root is missing

Reply:
`Root` always passes as this is called through [`T::ParamsOrigin::ensure_origin_or_root(origin)`](https://github.com/paritytech/polkadot-sdk/blob/53b7df05a58363119bac671a9173922e10724b18/substrate/frame/core-fellowship/src/lib.rs#L331) which always allows `Root`. Maybe worth explaining in the comment, what do you think of:

```rust
// Parameters are set by any of:
// - Root (implicitly allowed);
// - the FellowshipAdmin origin (i.e. token holder referendum);
// - a vote among all Head Ambassadors.
```

Reply:
Hmm on second thoughts maybe this is misleading, as somebody might (justifiably) call `ensure_origin()` and expect root to pass because of the comment.

Reply:
Resolved in [eff453a](https://github.com/paritytech/polkadot-sdk/pull/2024/commits/eff453a5f6820c2f16c20dc195becd79873e7efb).